src/paginationCalls/pageDescriptor.ts

```diff
@@ -103,6 +116,108 @@ export class PageDescriptor implements Descriptor {
     return stream;
   }
 
+  /**
+   * Create an async iterable which can be recursively called for data on-demand.
+   */
+  asyncIterate(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
+    const iterable = this.createIterator(options);
+    const funcPromise =
+      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
+    funcPromise
+      .then((func: GaxCall) => {
+        this.resolveParams(request, func, options);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+    return iterable;
+  }
+
+  createIterator(options: CallSettings): AsyncIterable<{} | undefined> {
+    const self = this;
+    const requestPageTokenFieldName = self.requestPageTokenField;
+    const responsePageTokenFieldName = self.responsePageTokenField;
+    const resourceField = self.resourceField;
+    const getNextPageRequest = self.getNextPageRequest;
+    const asyncIterable = {
+      [Symbol.asyncIterator]() {
+        const funcPromise = new Promise((resolve, reject) => {
+          self.resolveFunction = resolve;
+        });
+        const requestPromise = new Promise((resolve, reject) => {
+          self.resolveRequest = resolve;
+        });
+        const cache: Array<{}> = [];
+        let nextPageRequest: RequestType | null = {};
+        let firstCall = true;
+        return {
+          async next() {
+            const ongoingCall = new call.OngoingCallPromise(options.promise);
+            const func = (await funcPromise) as SimpleCallbackFunction;
+            const request = (await requestPromise) as RequestType;
+            if (cache.length > 0) {
+              const value = cache.shift();
+              return Promise.resolve({done: false, value});
+            }
+            if (!firstCall && !nextPageRequest) {
+              return Promise.resolve({done: true, value: -1});
```

Comment:
The `value` for `done: true` should be `undefined` (not `-1`) here and below.

Reply:
sure, thanks